common/connection_test.go

```diff
@@ -30,14 +30,14 @@ import (
 	"github.com/chromedp/cdproto/cdp"
 	"github.com/chromedp/cdproto/target"
 	"github.com/gorilla/websocket"
-	"github.com/grafana/xk6-browser/testutils"
+	"github.com/grafana/xk6-browser/common/test"
```

Comment:
Using pkg names like `utils`, `helper`, `common` is an anti-pattern in Go. In that aspect, everything is a helper or util! And, using `t.Helper` in a test function doesn't justify the reason that it should be called a helper 😅

However, as you said, we don't need to create a new package for these, but I didn't want to pollute the package namespace. But if we put them into a `_test` file we won't pollute the non-test namespace. Instead of `common/helper_tests.go`, I prefer putting them into a file called `common/protocol_test.go`.

Reply:
Sure, `common/protocol_test.go` SGTM. You really want to avoid naming anything "util" or "helper" :smile:

Like I said in the other comment, with this change you've just removed "utils" from the name, not fixed the real problem of having unrelated code in a single package. Having different files is a good thing, but with a few functions and hundred lines of code, we really don't know what the best way to structure this code is, and I'd argue we shouldn't worry about it yet.

At this point the only separation that _would_ be beneficial is separating actual test code from, well, test "helper" code. Even `common/protocol_test.go` would be confusing if we eventually have a `common/protocol.go` file, as that should be where its tests live...

But discussing this is taking more effort than whatever change we do, so `common/protocol_test.go` is better than a `common/test` package that has no tests.

Reply:
😄 Yeah, for now, I think it's OK too. In the future, if we ever going to have a `protocol_test.go` for an imaginary `Protocol` type, then we can move these in there. Those tests are only used by `connection` and `session` types which are protocol-related things. Maybe those two should be the same! As you can see, naming tells us where to look next :))